src/main/java/com/enderio/base/common/integrations/jei/EnderIOJEI.java

```diff
@@ -49,6 +55,17 @@ public void registerRecipes(IRecipeRegistration registration) {
 
         registration.addIngredientInfo(new ItemStack(EIOItems.GRAINS_OF_INFINITY.get()), VanillaTypes.ITEM_STACK, EIOLang.JEI_GRAINS_HAND_GRIND);
         registration.addIngredientInfo(new ItemStack(EIOItems.POWDERED_COAL.get()), VanillaTypes.ITEM_STACK, EIOLang.JEI_COAL_HAND_GRIND);
+
+        List<ItemStack> spawners = BrokenSpawnerItem.getPossibleStacks();
+        registration.getIngredientManager().removeIngredientsAtRuntime(VanillaTypes.ITEM_STACK, spawners);
+
+        List<ItemStack> glasses = new ArrayList<>();
+        for (GlassBlocks glass : EIOBlocks.GLASS_BLOCKS.values()) {
+            for (var color : glass.COLORS.values()) {
+                glasses.add(new ItemStack(color.asItem()));
+            }
+        }
+        registration.getIngredientManager().removeIngredientsAtRuntime(VanillaTypes.ITEM_STACK, glasses);
```

Comment:
I think we need a different way of hiding these - removing the ingredients is breaking recipes:
- coloured glass recipes show the clear glass as the result
- soul binding with broken spawner shows no mob on the input stack

Reply:
Verified the first one, the second one I'm unsure what you mean